rio_toa/reflectance.py

```diff
@@ -45,32 +53,61 @@ def reflectance(img, MR, AR, E, src_nodata=0):
         float32 ndarray with shape == input shape
 
     """
-    if E > np.finfo(float).eps:
-        rf = ((MR * img.astype(np.float32)) + AR) / np.sin(np.radians(E))
-        rf[img == src_nodata] = 0.0
-        return rf
+    rf = ((MR * img.astype(np.float32)) + AR) / np.sin(np.deg2rad(E))
+    rf[img == src_nodata] = 0.0
+
+    return rf
 
-    else:
-        raise ValueError(E, img.astype(np.float))
 
-def _reflectance_worker(data, window, ij, g_args):
+def _reflectance_worker(open_files, window, ij, g_args):
     """rio mucho worker for reflectance
     TODO
     ----
     integrate rescaling functionality for
     different output datatypes
     """
+    data = riomucho.utils.array_stack(
+            [src.read(window=window).astype(np.float32)
+                for src in open_files])
+
+    M_stack = np.ones(data.shape)
```

Comment:
@virginiayung I don't think we need to make a whole array for these variables - a single value for depth axis should suffice?